content/en/docs/features/signed-images.md

```diff
@@ -8,4 +8,145 @@ tags:
 - images
 ---
 
-TODO
+
+
+# Overview
+[Encrypted images](/docs/features/encrypted-images/) provide confidentiality,
+but they do not provide _authenticity_. Image signatures provide this additional
+property, preventing certain types of image tampering,
+[for example](https://docs.sigstore.dev/about/overview/#why-cryptographic-signing).
+
+In this brief guide, we show two tools that can be used to sign container images:
+[cosign](https://github.com/sigstore/cosign) and
+[skopeo](https://github.com/containers/skopeo). The skopeo approach is often
+called "simple signing" in the CoCo project.  In both cases, the general
+approach is to (1) create keys for signing, (2) sign a newly tagged image, and
+(3) update the [KBS](/docs/attestation/key-broker-service) with the public
+signature key and a security policy.
+
+
+# Creating Keys
+
+## cosign
+To generate a public-private keypair with cosign, provide your
+`COSIGN_PASSWORD` and use the `generate-key-pair` action:
+```shell
+$ COSIGN_PASSWORD=just1testing2password3 cosign generate-key-pair
+```
+
```

Comment:
You may want to mention about the files i.e. `cosign.key` and `cosign.pub` being generated here as these are being used in later steps.